src/dynamics/mpas/dyn_comp.F90

```diff
@@ -251,9 +254,9 @@ subroutine dyn_init(cam_runtime_opts, dyn_in, dyn_out)
                 ! Perform default initialization for all constituents.
                 ! Subsequently, they can be overridden depending on the namelist option (below) and
                 ! the actual availability (checked and handled by MPAS).
-                call dyn_debug_print('Calling set_default_constituent')
+                call dyn_debug_print('Calling dyn_exchange_constituent_state')
 
-                call set_default_constituent()
+                call dyn_exchange_constituent_state('e', .true., .false.)
```

Comment:
On a different note, after experimenting with the SE dycore we realized that the constituents object may not have all of the values set correctly at this point, with the assumption that the dycore would read in the values from the initial conditions (IC) file, and if not on the file just set them to zero.  You can find an example of this in the SE dycore interface for CAM-SIMA starting here:

https://github.com/ESCOMP/CAM-SIMA/blob/development/src/dynamics/se/dyn_comp.F90#L1496

This probably isn't necessary for this PR given that we'll almost certainly be using only analytic ICs for quite awhile, but I wanted to give you a heads up just in case.  If it would help then I am also happy to make an issue for it in CAM-SIMA so that we remember to tackle it later, just let me know!

Reply:
> I think it would help readability here if you passed in the arguments by name:

Addressed in https://github.com/ESCOMP/CAM-SIMA/pull/304/commits/122c64f642f3ddb211990a4f5fb6bc93f8617e98.

Reply:
Thanks for the explanation!  I agree that performing a conversion isn't needed here if we are just passing the default values, and if MPAS itself will handle the initialization of constituent data via input files using its own input stream methodology then I think we are ok.

I mostly was concerned if MPAS was going to try an integration step with some constituent inputs having a value of `HUGE(1.0_kind_phys)`, but it sounds like that shouldn't be the case as long as we have the proper input files or default values configured on the physics-side.

I'll thus go ahead and mark this issue as resolved.  If we do find that "bad" default values are sneaking in then we can always add an additional check for it later.  Thanks again for the info!